runtime/tr.source/trj9/codegen/J9TreeEvaluator.hpp

```diff
@@ -54,6 +54,14 @@ class OMR_EXTENSIBLE TreeEvaluator: public OMR::TreeEvaluatorConnector
       ClassEqualityTest,                // Needs object class: y, needs cast class: y
       SuperClassTest,                   // Needs object class: y, needs cast class: y
       CastClassCacheTest,               // Needs object class: y, needs cast class: y
+      /* Currently following two tests are only made available for z Systems
+       * to make sure consumer of this API on other platform do not need to
+       * make any changes in their implementations.
+       */
```

Comment:
z Systems have rebranded to IBM Z. We should use the new name if possible. Also why do we need the `ifdef` below? Typically we try to avoid platform specific `ifdefs` in platform agnostic code. Can we do without it?

Reply:
@gita-omr @0dvictor FYI this is relevant to Power and x86.

Reply:
`x86` does not use the new API to generate inlined tests for instanceOf/checkCast node.
I had this tests added in common API because in future `x86` and `P` can take advantage of it.  
Only reason I have this guarded for `IBM Z` is because `P` uses this API and atleast if they are not generating this test, they need to handle it in their evaluator.

Reply:
I don't think I'm quite following. Why do they need to handle it in their evaluator if we never generate such tests on Power?

Reply:
So you are suggesting to use Platform query instead of `#IFDEF` ?

Reply:
I'm suggesting we remove platforms specific `ifdefs` in the platform agnostic code generator and figure out what changes if any are needed to make the Power code generator be able to handle the new definitions within this `enum`. I suspect if any changes are needed they are as simple as adding `case` labels to `switch` statements with assertions. I don't think we should have to make any changes though but I am not as familiar with this code as you may be.